relayer/chains/mock/mock_chain_processor.go

```diff
@@ -0,0 +1,171 @@
+package mock
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/cosmos/relayer/v2/relayer/ibc"
+	"github.com/cosmos/relayer/v2/relayer/paths"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+
+	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"go.uber.org/zap"
+)
+
+const (
+	minQueryLoopDuration = 1 * time.Second
```

Comment:
It would be nice to eventually make this configurable from the config file, I don't think it's necessary here though.

Reply:
Outside the mock chain processor I was thinking this could potentially be detected from the chain block speed. If comparing to a sample rate of 1/(2n +1), then we can measure the delta t between blocks when it initially starts, divide by 2, and subtract a small percentage, and use that as the minQueryLoopDuration. Then we will be doing 2 simple block height queries per block but will be quick enough after the block is produced to handle it.

Another idea is to measure the block time and have it target the exact time that it would expect the next block to be available.

Reply:
Oh yeah I dig this!! Removing the burden from operators and dynamically calculating this value would be nice.

Reply:
This is the type of nice to have feature that makes the whole thing run more smoothly